src/components/tl/ucp/tl_ucp_coll.c

```diff
@@ -405,19 +462,24 @@ ucc_status_t ucc_tl_ucp_coll_dynamic_segment_exchange(ucc_tl_ucp_task_t *task)
     return status;
 }
 
-void ucc_tl_ucp_coll_dynamic_segment_finalize(ucc_tl_ucp_task_t *task)
+ucc_status_t ucc_tl_ucp_coll_dynamic_segment_finalize(ucc_tl_ucp_task_t *task)
 {
     ucc_tl_ucp_team_t    *tl_team = UCC_TL_UCP_TASK_TEAM(task);
     ucc_tl_ucp_context_t *ctx     = UCC_TL_UCP_TEAM_CTX(tl_team);
     int                   i       = 0;
     int                   j       = 0;
+    ucs_status_t          status;
     /* free library resources, unmap user resources */
     if (ctx->dyn_seg_buf) {
         /* unmap and release packed buffers */
         for (i = 0; i < ctx->n_dynrinfo_segs; i++) {
             if (ctx->dynamic_remote_info[i].mem_h) {
-                ucp_mem_unmap(ctx->worker.ucp_context,
+                status = ucp_mem_unmap(ctx->worker.ucp_context,
                               ctx->dynamic_remote_info[i].mem_h);
+                if (UCS_OK != status) {
+                    tl_error(UCC_TL_UCP_TEAM_LIB(tl_team), "Failed to unmap memory");
+                    return ucs_status_to_ucc_status(status);
```

Comment:
nit: shouldn't we, even in the case of an error, still try to free and release the remaining resource?

Reply:
done